src/me/mrCookieSlime/Slimefun/Objects/tasks/RainbowTicker.java

```diff
@@ -49,7 +49,7 @@ else if (MaterialHelper.isTerracotta(b.getType())){
 
 	@Override
 	public void uniqueTick() {
-		index = ((index == queue.length - 1) ? 0: index + 1);
+		index = ((index == queue[queue.length - 1]) ? 0: index + 1);
```

Comment:
Why are you comparing an index value with a data value here?
Please elaborate because I can see this causing a ton of problems.

Reply:
Tbh I don't know either how this is fixing the issue but I was doing trying stuff to see if it fixes the issue and magically this did.

Reply:
It shouldn't though. So I am leaving it open for now.

Reply:
After a thought it worked because when I tested, I tested it for rainbow blocks which had these values in queue:
0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15

When index needed to change its value was 15 which is also the last value in queue.
In the older version index became values like 767 for some reason which caused the ArrayIndexOutOfBounds exception in 53rd line.

Reply:
Just a theory, changing the == with >= in older code may also fix this.
Or adding a space before ":"

Reply:
>= would have also been my suggestion to fix it.
But adding a space there won't do anything.